libsolidity/formal/SMTEncoder.cpp

```diff
@@ -893,6 +896,18 @@ void SMTEncoder::visitGasLeft(FunctionCall const& _funCall)
 		m_context.addAssertion(symbolicVar->currentValue() <= symbolicVar->valueAtIndex(index - 1));
 }
 
+void SMTEncoder::visitBlobHash(FunctionCall const& _funCall)
+{
+	constexpr unsigned BLOB_TRANSACTION_LIMIT = 9; // Since pectra
+	auto indexExpr = expr(*_funCall.arguments().at(0));
```

Comment:
Wouldn't this throw an out-of-bounds error if there are no arguments? Or is it guaranteed at this point that the function call has at least one argument? Either way, I think it wouldn't hurt to assert that beforehand.

Reply:
The compiler reports an error if you pass wrong number of arguments to `blobhash` and reject the source code even before SMTChecker runs.
So I think in this case we can rely on the fact there will be exactly one argument.
We do the same for `blockhash`, there is also no check there.

I can add it if you really insist, but in this case I really think it is redundant.
I guess you mean `solAssert`, and not basic `assert`, right?

Reply:
> I can add it if you really insist, but in this case I really think it is redundant. I guess you mean `solAssert`, and not basic `assert`, right?

 Yes, that was what a mean. But you are right, it seems fine.